### PR TITLE
feat(linter): eslint-plugin-jest/prefer-comparison-matcher

### PR DESCRIPTION
**Pull request by eryue0220**
Rule Detail: [link](https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/prefer-comparison-matcher.ts)

### CODE REVIEW

File: crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+        let preferred_matcher = Self::determine_matcher(binary_expr.operator, negated);
+        let Some(prefer_matcher_name) = preferred_matcher else {
+            return;
+        };
+
+        ctx.diagnostic_with_fix(UseToBeComparison(prefer_matcher_name, matcher.span), || {
+            let call_span_end = Span::new(binary_expr.span.end, parent_call_expr.span.end)
+                .source_text(ctx.source_text());
+            let arg_span_end = Span::new(matcher_arg_value.span.end, call_expr.span.end)
+                .source_text(ctx.source_text());
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs:129]: Nit: can we add some comments to describe where `call_span_end` and `arg_span_end` point to? e.g.

```js
expect(1 > 2).toBe(false)
            ^
```

File: crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+            .modifiers()
+            .iter()
+            .filter(|modifier| modifier.is_name_equal("not"))
+            .count()
+            > 0;
+        let Expression::BooleanLiteral(matcher_arg_value) =
+            first_matcher_arg.get_inner_expression()
+        else {
+            return;
+        };
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs:118]: Seems there are duplicate check in line 103.

File: crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+        content.print_str(prefer_matcher_name.as_bytes());
+        content.print(b'(');
+        content.print_expression(&binary_expr.right);
+        content.print_str(arg_span_end.as_bytes());
+        content.into_source_text()
+    }
+}
+
+#[test]
+fn test() {
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs:212]: These test cases also need many codes.😢

File: crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+    /// ### Examples
+    ///
+    /// ```js
+    /// // invalid
+    /// expect(x > 5).toBe(true);
+    /// expect(x < 7).not.toEqual(true);
+    /// expect(x <= y).toStrictEqual(true);
+    /// ```
+    ///
+    /// ```js /// // valid
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs:51]: A small typo. Maybe the "// valid" is in next line

File: crates/oxc_linter/src/rules/jest/prefer_called_with.rs [hunk 5]
```diff
@@ -68,15 +68,12 @@ impl PreferCalledWith {
...
-                    break;
-                }
-            }
-        }
+        let has_not_modifier = jest_fn_call
+            .modifiers()
+            .iter()
+            .filter(|modifier| modifier.is_name_equal("not"))
+            .count()
+            > 0;
```
Comment (bschoenmaeckers) [inline crates/oxc_linter/src/rules/jest/prefer_called_with.rs:76]:
```suggestion
            .any(|modifier| modifier.is_name_equal("not"));
```

File: crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs (hunk 2 above)
Comment (eryue0220) [reply crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs:114]: Done, and I removed the check in line 103.

File: crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs (hunk 4 above)
Comment (eryue0220) [reply crates/oxc_linter/src/rules/jest/prefer_comparison_matcher.rs]: Done. Nice Catch.

File: crates/oxc_linter/src/rules/jest/prefer_called_with.rs (hunk 5 above)
Comment (eryue0220) [reply crates/oxc_linter/src/rules/jest/prefer_called_with.rs]: Done, thank you. And I also upgrade the same usages in others rules.

**Review by mysteryven — APPROVED** [review]
Nice!

**Comment by mysteryven** [comment]
There are some test cases failed.

**Comment by eryue0220** [comment]
> There are some test cases failed.

I will upgrade it later.

**Pull request merged**